eth/filters/bor_api.go

```diff
@@ -67,7 +67,7 @@ func (api *PublicFilterAPI) NewDeposits(ctx context.Context, crit ethereum.State
 		for {
 			select {
 			case h := <-stateSyncData:
-				if crit.ID == h.ID || bytes.Compare(crit.Contract.Bytes(), h.Contract.Bytes()) == 0 ||
+				if h != nil || crit.ID == h.ID || bytes.Equal(crit.Contract.Bytes(), h.Contract.Bytes()) ||
 					(crit.ID == 0 && crit.Contract == common.Address{}) {
```

Comment:
It is wrong. '||' is OR. So if h is nil it will occur panic again on checking `crit.ID == h.ID` statement.
For example: https://go.dev/play/p/a9GdV-uwc9N
my suggestion:
```
	if h != nil && (crit.ID == h.ID || 
		bytes.Compare(crit.Contract.Bytes(), h.Contract.Bytes()) == 0 ||
		(crit.ID == 0 && crit.Contract == common.Address{})) {
					notifier.Notify(rpcSub.ID, h)
	}
```

Reply:
```suggestion
				if h != nil && (crit.ID == h.ID || 
					bytes.Compare(crit.Contract.Bytes(), h.Contract.Bytes()) == 0 ||
					(crit.ID == 0 && crit.Contract == common.Address{})) {
```

Reply:
Resolved. Thanks!